system_tests/manager/test_hostpool_plugin.py

```diff
@@ -13,193 +13,168 @@
+                'ubuntu_image_id': attributes.ubuntu_14_04_image_name,
+                'flavor_id': attributes.medium_flavor_name,
+                'floating_network_id': attributes.floating_network_id,
+                'network_name': attributes.network_name,
+                'key_pair_name': attributes.keypair_name,
+                'private_key_path': self.manager.remote_private_key_path,
+                # key_path is where the hostpool service host will store the
+                # key to the hosts; it is not the same as the agent key
+                'key_path': '/tmp/' + str(uuid.uuid4()),
+            }
+        return self._inputs
+
+    def verify_installation(self):
+        """Check that 3 hosts are added to the pool, but not used. Then,
+        scale up, and check that another host has been added.
+        """
+        super(HostPoolExample, self).verify_installation()
+        hosts = self.get_hosts()
+        self.assertEqual(len(HOST_TEMPLATES), len(hosts))
         for host in hosts:
-            self._assert_host_state(host)
+            self.assert_host_state(host, allocated=False)
         self._scale(UBUNTU_HOST)
-        hosts = self._get_hosts(self.host_pool_service_deployment_id)
-        self.assertEquals(len(hosts), len(HOST_TEMPLATES) + 1)
-        self._test_nodecellar_impl('host-pool-blueprint.yaml')
-        hosts = self._get_hosts(self.host_pool_service_deployment_id)
-        self.assertEquals(len(hosts), len(HOST_TEMPLATES) + 1)
-        for host in hosts:
-            self._assert_host_state(host)
+        hosts = self.get_hosts()
+        self.assertEqual(len(HOST_TEMPLATES) + 1, len(hosts))
 
+    def get_hosts(self):
+        """Hosts added to the pool"""
+        endpoint_url = self.get_endpoint_url()
+        response = requests.get('{0}/hosts'.format(endpoint_url))
+        return response.json()
 
-    def _test_nodecellar_impl(
-        self, blueprint_file, execute_timeout=DEFAULT_EXECUTE_TIMEOUT
-        ):
-
-        self.repo_dir = clone(self.repo_url, self.workdir)
-        self.blueprint_yaml = self.repo_dir / blueprint_file
-
-        self.modify_blueprint()
-
-        before, after = self.upload_deploy_and_execute_install(
-            inputs=self.get_inputs(),
-            execute_timeout=execute_timeout
+    def get_endpoint_url(self):
+        """Address of the hostpool service API"""
+        return 'http://{0}:{1}'.format(
+            self.outputs['endpoint']['ip_address'],
+            self.outputs['endpoint']['port']
         )
 
-        self.post_install_assertions(before, after)
-
-        self.execute_uninstall()
-
-        self.post_uninstall_assertions()
-
-    def post_install_assertions(self, before_state, after_state):
-
-        delta = self.get_manager_state_delta(before_state, after_state)
-
-        self.logger.info('Current manager state: {0}'.format(delta))
-
-        self.assertEqual(len(delta['blueprints']), 1,
```

Comment:
asserting on blueprints, deployments, executions, etc... is not necessary - we're not testing the core execution engine here! And anyway, the `AbstractExample` superclass does some basic assertions.